### PR TITLE
adding in user profiles analytics for signup flow from public library/profile

### PR DESCRIPTION
**Pull request by ahsu1230**
I know you're doing tracking for user profiles and there's a bit of overlap
@yipingliao

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/signup/signupService.js [hunk 1]
```diff
@@ -122,97 +125,26 @@ angular.module('kifi')
...
     };
 
     $scope.fbAuthFromLibrary = function () {
-      if ($scope.userData.libraryId) {
-        $rootScope.$emit('trackLibraryEvent', 'click', { type: 'signupLibrary', action: 'clickAuthFacebook' });
-      }
-      //return $scope.fbAuth(); //unused
-    };
-
-    $scope.fbAuth = function () {
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/signup/signupService.js:154]: unused code with one-step FB clicking

File: kifi-backend/modules/shoebox/angular/src/signup/signupService.js [hunk 2]
```diff
@@ -122,97 +125,26 @@ angular.module('kifi')
...
-        $rootScope.$emit('trackLibraryEvent', 'click', { type: 'signupLibrary', action: 'clickSignUpButton' });
-      }
+      emitTracking('click', 'signup', { action: 'clickSignUpButton' });
 
       modalService.close();
       $scope.userData.method = 'email';
       registerFinalizeModal();
     };
 
-    $scope.fbInit = function () {
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/signup/signupService.js:135]: unused code with one-step FB route

File: kifi-backend/modules/shoebox/angular/src/signup/signupService.js [hunk 3]
```diff
@@ -75,43 +75,46 @@ angular.module('kifi')
...
         template: 'signup/registerModal.tpl.html',
         scope: $scope
       }), function onClose() {
         if (!$scope.emailSubmitted) { // did not submit email, so closed modal
-          $analytics.eventTrack('visitor_clicked_page', {type: 'signupLibrary', action: 'close'});
-        }
-      });
-    };
-
-    var facebookLogin = function () {
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/signup/signupService.js:103]: unused now with FB one click

File: kifi-backend/modules/shoebox/angular/src/userProfile/userProfile.js [hunk 4]
```diff
@@ -110,6 +110,20 @@ angular.module('kifi')
     });
     $scope.$on('$destroy', deregisterCurrentLibrary);
 
+    var deregisterTrackUserProfile = $rootScope.$on('trackUserProfileEvent', function (e, eventType, attributes) {
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/userProfile/userProfile.js:113]: trying to emulate how `trackLibraryEvent` in `library.js` works. Would this work with you? @yipingliao

File: kifi-backend/modules/shoebox/angular/src/signup/signupService.js [hunk 5]
```diff
@@ -75,43 +87,46 @@ angular.module('kifi')
...
+        $analytics.eventTrack(eventName, {type: typeBase, action: action});
+      } else {
+        typeBase += 'UserProfile';
+        $analytics.eventTrack(eventName, {type: typeBase, action: action});
+      }
+    }
 
-    var registerModal = function () {
+    function emitTracking(eventType, typeBase, attributes) {
       if ($scope.userData.libraryId) {
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/signup/signupService.js]: Are we trying to determine what page we're on with these tests?

File: kifi-backend/modules/shoebox/angular/src/userProfile/userProfile.js (hunk 4 above)
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/userProfile/userProfile.js:113]: Thanks! Will come in handy.

File: kifi-backend/modules/shoebox/angular/src/signup/signupService.js (hunk 5 above)
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/signup/signupService.js]: thanks @yipingliao for the tip with $state! So useful

**Pull request merged**